test/Microsoft.AspNet.Security.Test/OAuthBearer/OAuthBearerMiddlewareTests.cs

```diff
@@ -0,0 +1,245 @@
+        }
+
+        private static Task SecurityTokenValidated(SecurityTokenValidatedNotification<HttpContext, OAuthBearerAuthenticationOptions> notification)
+        {
+            List<Claim> claims =
+                new List<Claim>
+                {
+                    new Claim(ClaimTypes.Email, "bob@contoso.com"),
+                    new Claim(ClaimsIdentity.DefaultNameClaimType, "bob"),
+                };
+
+            notification.AuthenticationTicket = new AuthenticationTicket(new ClaimsIdentity(claims, notification.Options.AuthenticationType), new Http.Security.AuthenticationProperties());
+            notification.HandleResponse();
+
+            return Task.FromResult<object>(null);
+        }
+
+        class BlobTokenValidator : ISecurityTokenValidator
+        {
+
+            public BlobTokenValidator(string authenticationType)
+            {
+                AuthenticationType = authenticationType;
+            }
+
+            public string AuthenticationType { get; set; }
+
+            public bool CanValidateToken
+            {
+                get
+                {
+                    return true;
+                }
+            }
+
+            public int MaximumTokenSizeInBytes
+            {
+                get
+                {
+                    return 2*2*1024;
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public bool CanReadToken(string securityToken)
+            {
+                return true;
+            }
+
+            public ClaimsPrincipal ValidateToken(string securityToken, TokenValidationParameters validationParameters, out SecurityToken validatedToken)
+            {
+                validatedToken = null;
+                List<Claim> claims =
+                    new List<Claim>
+                    {
+                                    new Claim(ClaimTypes.Email, "bob@contoso.com"),
+                                    new Claim(ClaimsIdentity.DefaultNameClaimType, "bob"),
+                    };
+
+                return new ClaimsPrincipal(new ClaimsIdentity(claims, AuthenticationType));
+            }
+        }
+
+        private static TestServer CreateServer(Action<OAuthBearerAuthenticationOptions> configureOptions, Func<HttpContext, bool> handler = null)
+        {
+            return TestServer.Create(app =>
+            {
+                app.UseServices(services =>
+                {
+                    services.AddDataProtection();
+                });
+                app.UseCookieAuthentication(options =>
+                {
+                    options.AuthenticationType = "Bearer";
+                });
+                if (configureOptions != null)
+                {
+                    app.UseOAuthBearerAuthentication(configureOptions);
+                }
+                app.Use(async (context, next) =>
```

Comment:
I don't understand this test, what does it validate? `next()` will just return 404 and any request to `/oauth` will return 200. Nothing checks the user identity, issues 401s, etc..